test/e2e/page-objects/pages/test-dapp.ts

```diff
@@ -289,25 +299,18 @@ class TestDapp {
   }) {
     console.log('Connect account to test dapp');
     await this.driver.clickElement(this.connectAccountButton);
-    await this.driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
-    await this.driver.waitForSelector(this.connectMetaMaskMessage);
     if (connectAccountButtonEnabled) {
-      await this.driver.clickElementAndWaitForWindowToClose(
-        this.confirmDialogButton,
-      );
+      await this.confirmConnectAccountModal();
     } else {
+      await this.driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+      await this.driver.waitForSelector(this.connectMetaMaskMessage);
       const confirmConnectDialogButton = await this.driver.findElement(
         this.confirmDialogButton,
       );
       assert.equal(await confirmConnectDialogButton.isEnabled(), false);
     }
-
     if (publicAddress) {
-      await this.driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
-      await this.driver.waitForSelector({
-        css: this.connectedAccount,
-        text: publicAddress.toLowerCase(),
-      });
+      await this.check_connectedAccounts(publicAddress);
```

Comment:
`publicAddress` is optional, should we be removing the condition check?

Reply:
yes, the `publicAddress` is optional. When we expect to connect an account to test dapp, we need to pass the address as a parameter so we can verify that the account is successfully connected. When we want to check that an account cannot connect to the test dapp (for example, a Bitcoin account), we don't need to pass the address, we just need to check that the connect button is disabled. Therefore, we need a condition check here.

Reply:
Ah I see the method takes a second param